mono/sgen/sgen-alloc.c

```diff
@@ -507,15 +508,20 @@ sgen_alloc_obj_mature (GCVTable vtable, size_t size)
 void
 sgen_clear_tlabs (void)
 {
+	guint64 total_bytes_allocated_globally = 0;
+
 	FOREACH_THREAD_ALL (info) {
 		/* A new TLAB will be allocated when the thread does its first allocation */
 		info->total_bytes_allocated += info->tlab_next - info->tlab_start;
+		total_bytes_allocated_globally += info->total_bytes_allocated;
 		info->tlab_start = NULL;
 		info->tlab_next = NULL;
 		info->tlab_temp_end = NULL;
 		info->tlab_real_end = NULL;
 	} FOREACH_THREAD_END
-}
+
```

Comment:
This doesn't account for threads that allocated and have died / detached.

This updates the counter only during a collection, so it is not precise. I also don't understand why would we do it this way, when for the thread specific counter we were already providing real time information. We could just iterate over all the threads during the icall and use the same computation from *mono_gc_get_allocated_bytes_for_current_thread* while keeping an incrementing counter for the allocation of threads that have died (I think in sgen_client_thread_detach_with_lock)

Reply:
So, based on the API here: https://docs.microsoft.com/en-us/dotnet/api/system.gc.gettotalallocatedbytes?view=netcore-3.0 . it takes a "precise" flag. If its true, you get an accurate count, but there may be a performance penalty. If false, its fast but maybe inaccurate.

I figured being accurate up to the last minor collection was good enough for precise=false.  Can I iterate over all the threads with out taking out a lock on the list of threads, thereby blocking new thread creation? 

I will definately fix the missing counts from dead/detached threads though.

Reply:
I didn't notice that the PR was not ready. I personally prefer to PR when the code is ready for final review. If you prefer to do it this way (jenkins might hate you), make sure to add WIP or Draft tag just to make it clear.

I'd say it's ok to have a counter updated at collection time and for precise stop the world and iterate all threads.

Reply:
I will do it that way in the future, but is there another way to share partial changes to discuss stuff?